CMakeLists.txt

```diff
@@ -6,7 +6,9 @@ cmake_minimum_required(VERSION 3.5.0)
 set(_msg "Setting C++ standard")
 message(STATUS "${_msg}")
 if(DEFINED Slicer_DIR)
-  set(CMAKE_CXX_STANDARD 14)
+  if(NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+  endif()
 endif()
 if(NOT DEFINED CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 98)
```

Comment:
Not a blocker to merging this change, but if a `CMAKE_CXX_STANDARD` is not set, dcmqi wants to still continue setting the `CMAKE_CXX_STANDARD` to `98` for a non-Slicer dir and to `14` for a Slicer dir? Does dcmqi want to standardize what C++ standard it should conform to?

Reply:
Would you recommend we switch to `14` in all places as default?

Reply:
The current Slicer stable will build as `14` while Slicer preview will build as `17`. If not a `Slicer_DIR`, then the minimum of Slicer stable and preview would seem to make sense. Therefore setting it to `14` when not a `Slicer_DIR` would make sense.

In terms of general relevancy, latest versions of many projects are now switching to `17`.

Reply:
Thank you @jamesobutler!

@jcfr let me know if you prefer to merge this PR and handle this separately, or want to update in place. Either is fine with me.